addons/fleet/models/fleet_vehicle.py

```diff
@@ -218,6 +218,11 @@ def write(self, vals):
         if 'driver_id' in vals and vals['driver_id']:
             driver_id = vals['driver_id']
             self.filtered(lambda v: v.driver_id.id != driver_id).create_driver_history(driver_id)
+            note = _('Specify the End date of %s') % self.driver_id.name
+            self.activity_schedule(
+                'mail.mail_activity_data_todo',
+                user_id=self.manager_id.id or self.env.user.id,
```

Comment:
self can be multi, you can put it in a for loop (write a driver on multi record is not common, but you must manage this use case)